lib/media/media_source_engine.js

```diff
@@ -921,12 +922,12 @@ shaka.media.MediaSourceEngine = class {
       }
     } else if (!mimeType.includes('/mp4') && !mimeType.includes('/webm') &&
         shaka.util.TsParser.probe(uint8ArrayData)) {
-      if (!this.tsParser_) {
-        this.tsParser_ = new shaka.util.TsParser();
+      if (!this.tsParser_[contentType]) {
+        this.tsParser_[contentType] = new shaka.util.TsParser();
       } else {
-        this.tsParser_.clearData();
+        this.tsParser_[contentType].clearData();
```

Comment:
So, I don't know the TS decoding process all that well. But I do have a concern.

If I am understanding correctly, the PMT packets are what stores the video and audio PIDs, right? If we get a PMT packet that has both a video and audio PID with PR, won't we end up with one of the two TS parsers having the other parser's PID and the other TS parser having no PIDs at all? Or does this change mean we will be parsing every PMT packet twice, I can't tell?

Reply:
Audio segment --> PID for audio 80
Video segment --> PID for video 80

What we will have is a TS parser for each contentType so we won't have problems in case different segments have the same PID for audio and video

Reply:
Okay, I think I see. I guess I had been assuming we were dealing with multiplexed content, but it wouldn't make much sense for multiplexed content to have video and audio with the same PID anyway. If this is two separate streams, it makes sense they would have separate PMT packets to parse.